install_files/ansible-base/roles/restrict-direct-access/templates/rules_v4

```diff
@@ -90,26 +94,20 @@
 {% endif %}
 
 
-# Permit direct SSH access for staging environment. Using a dual conditional here for safety;
-# the role default var "allow_direct_access" is false, and the "staging" name is not present in prod.
-# Both conditions must be satisfied to add these exemptions.
-{% if allow_direct_access == true and
-      'staging' in group_names %}
-
-{% if 'securedrop_application_server' in group_names %}
--A INPUT -i {{ ansible_default_ipv4.interface }}  -p tcp --dport 22 -m state --state NEW,ESTABLISHED,RELATED -j ACCEPT
--A OUTPUT -o {{ ansible_default_ipv4.interface }}  -p tcp -m owner --uid-owner root --sport 22 -m state --state NEW,ESTABLISHED,RELATED -j ACCEPT
--A OUTPUT -p udp --dport 53 -m state --state NEW,ESTABLISHED,RELATED -j ACCEPT
--A INPUT -p udp --sport 53 -m state --state NEW,ESTABLISHED,RELATED -j ACCEPT
-
-{% elif 'securedrop_monitor_server' in group_names %}
--A INPUT -i {{ ansible_default_ipv4.interface }} -p tcp --dport 22 -m state --state NEW,ESTABLISHED,RELATED -j ACCEPT
--A OUTPUT -o {{ ansible_default_ipv4.interface }} -p tcp --sport 22 -m state --state NEW,ESTABLISHED,RELATED -j ACCEPT
--A OUTPUT -p udp --dport 53 -m state --state NEW,ESTABLISHED,RELATED -j ACCEPT
--A INPUT -p udp --sport 53 -m state --state NEW,ESTABLISHED,RELATED -j ACCEPT
-{% endif %}
-
-{% endif %}
+# Permit direct SSH access.
+# Allowed for staging and optionally for production (disables ssh over tor)
+{% for interface in ansible_interfaces -%}
+  {%- set int_details = hostvars[inventory_hostname]['ansible_'+interface].ipv4 -%}
+  {%- set net_string = '/'.join([int_details.network,int_details.netmask]) -%}
+  {%- if ssh_ip|ipaddr(net_string) -%}
+    {%- set ssh_local_net = net_string|ipaddr('cidr') -%}
+    {%- set ssh_int = interface -%}
+      {%- if enable_ssh_over_tor == true or 'staging' in group_names -%}
```

Comment:
The `if enable_ssh_over_tor` conditional should be the governor of whether the entire "Permit direct SSH access" block is entered. Let's move this to the top of the block and iterate over the interfaces only if required.

The `netmask` attributes on ifaces will also require additional pip requirements, but I see you already added those to the admin requirements file.

Reply:
unfortunately -- if you do that logic (only loop over this `if enable_ssh_over_tor`) you are unable to use these values outside that loop. That bit me hard while debugging and I couldnt find a way to set jinja variables global inside a loop

Reply:
for reference of similar issue I ran into when trying to add this under a giant conditional at the top previously -- https://stackoverflow.com/questions/17925674/jinja2-local-global-variable

i did just find a post of someone assigning to a dictionary instead though to get around this --> https://stackoverflow.com/questions/9486393/jinja2-change-the-value-of-a-variable-inside-a-loop .

I honestly dont think its a huge deal as-is. can you chip in with more context on why you think the current logic is going to be problematic?